jscomp/others/belt_Option.ml

```diff
@@ -0,0 +1,27 @@
+let getExn = function
+  | Some x -> x
+  | None -> assert false
+
+let fold opt default f = match opt with
```

Comment:
The name will need some thinking

Reply:
People from the JS world probably know this as `reduce` or something along those lines at least for Lists/Arrays. We use fold internally because we're used to it from Scala/Haskell.